app/tests/spec/models/reliers/oauth.js

```diff
@@ -233,6 +233,14 @@ define(function (require, exports, module) {
           testInvalidQueryParams('access_type', invalidValues);
         });
 
+        describe('action', function () {
+          var validValues = [undefined, 'email', 'signin', 'signup', 'force_auth'];
+          testValidQueryParams('action', validValues, 'action', validValues);
+
+          var invalidValues = ['', ' ', 'invalid'];
+          testInvalidQueryParams('action', invalidValues);
+        });
+
```

Comment:
The new `login_hint` query param also needs some tests in this file.

Reply:
@rfk what should we test for login_hint ?

Reply:
Just that it accepts something that looks like an email address, and rejects something that doesn't look like an email address.